dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java

```diff
@@ -1808,7 +1812,8 @@ public void submitStandByTask() throws StateEventHandleException {
                 getPreVarPool(task, preTask);
             }
             DependResult dependResult = getDependResultForTask(task);
-            if (DependResult.SUCCESS == dependResult) {
+            if (DependResult.SUCCESS == dependResult || ((DependResult.WAITING != dependResult)
```

Comment:
What about changing it to `if ((DependResult.WAITING != dependResult)
                    && (processInstance.getFailureStrategy() == FailureStrategy.CONTINUE))` ? @ruanwenjun